packages/utils/upgrade/resources/codemods/5.0.0/comment-out-lifecycle-files.code.ts

```diff
@@ -0,0 +1,62 @@
+import { Transform } from 'jscodeshift';
+
+/**
+ * comments out lifecycles.js/ts files and adds a description for the reason at the top
+ */
+const transform: Transform = (file, api) => {
+  const j = api.jscodeshift;
+  const root = j(file.source);
+
+  if (file.path.endsWith('lifecycles.js') || file.path.endsWith('lifecycles.ts')) {
+    // Get the entire source code as a string
+    const sourceCode = root.toSource();
+
+    // Split the source code into lines and prepend // to each line
+    // we are use line comments instead of block comments so we don't face issues with existing block comments
+    const commentedCode = sourceCode
+      .split('\n')
+      .map((line) => `// ${line}`)
+      .join('\n');
+
+    // Add a header comment at the top to explain why the file is commented out
+    const headerComment = `
+/*
+ *
+ * ============================================================
+ * WARNING: THIS FILE HAS BEEN COMMENTED OUT
+ * ============================================================
+ *
+ * CONTEXT:
+ *
+ * The lifecycles.js file has been commented out to prevent unintended side effects when starting Strapi 5 for the first time after migrating to the document service.
+ *
+ * STRAPI 5 introduces a new document service that handles lifecycles differently compared to previous versions. Without migrating your lifecycles to document service middlewares, you may experience issues such as:
+ *
+ * - \`unpublish\` actions triggering \`delete\` lifecycles for every locale with a published entity, which differs from the expected behavior in v4.
+ * - \`discardDraft\` actions triggering both \`create\` and \`delete\` lifecycles, leading to potential confusion.
+ *
+ * MIGRATION GUIDE:
+ *
+ * For a thorough guide on migrating your lifecycles to document service middlewares, please refer to the following link:
+ * [Document Services Middlewares Migration Guide](https://docs-next.strapi.io/dev-docs/migration/v4-to-v5/breaking-changes/lifecycle-hooks-document-service)
```

Comment:
We'll need to update that with `docs.strapi.io` instead of `docs-next`. Do we want to do it now or right before the release?

Reply:
_(I would personally vote to link the stable doc right now so that we don't forget)_

Reply:
we certainly can do it now, It'll just be not working for now, the only concern would be people who are migrating now

Reply:
I think it's fine if the doc link isn't available during RC, I prefer to be sure we have a valid link upon release so that the vast majority of users don't have issues :slightly_smiling_face: